File: ulohy/32-zrcadlo/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let w;
let h;
let direction;

const paper = [];

let rowsLoaded = 0;
let colsLoaded = 0;

rl.on("line", (line) => {
    if (w === undefined) {
        w = parseInt(line);
        return;
    }

    if (h === undefined) {
        h = parseInt(line);
        for (let i = 0; i < h; i++) {
            paper.push([]);
        }

        if (h < 1 || w < 1) {
            rl.close();
        }
        return;
    }

    if (direction === undefined) {
        direction = line;
        return;
    }

    if (line === "") {
        return;
    }

    paper[rowsLoaded].push(line);

    colsLoaded++;
    if (colsLoaded >= w) {
        rowsLoaded++;
        colsLoaded = 0;
    }

    if (rowsLoaded < h) {
        return;
    }

    switch (direction) {
        case "p":
            for (let row = 0; row < h; row++) {
                for (let col = w - 1; col >= 0; col--) {
                    process.stdout.write(paper[row][col]);
                }
                process.stdout.write("\n");
            }
            break;

        case "s":
            for (let row = h - 1; row >= 0; row--) {
                for (let col = 0; col < w; col++) {
                    process.stdout.write(paper[row][col]);
                }
                process.stdout.write("\n");
            }
            break;
    }

    rl.close();
});

File: ulohy/32-zrcadlo/reseni/main.cpp
#include <iostream>
#include <vector>

int main() {
    int w, h;
    char direction;

    std::cin >> w >> h >> direction;

    if (w < 1 || h < 1) {
        return 0;
    }

    std::vector<std::vector<char>> paper(h, std::vector<char>(w));

    for (int row = 0; row < h; row++) {
        for (int col = 0; col < w; col++) {
            std::cin >> paper[row][col];
        }
    }

    switch (direction) {
        case 'p':
            for (int row = 0; row < h; row++) {
                for (int col = w - 1; col >= 0; col--) {
                    std::cout << paper[row][col];
                }
                std::cout << std::endl;
            }
            break;

        case 's':
            for (int row = h - 1; row >= 0; row--) {
                for (int col = 0; col < w; col++) {
                    std::cout << paper[row][col];
                }
                std::cout << std::endl;
            }
            break;
    }

    return 0;
}
